### ISSUE TITLE
Recursive scan of local project

### ISSUE THREAD

**Issue by dahankzter** [src-d/go-license-detector#43]
Perhaps I am doing something wrong but GLD does not seem to recursively scan may vendor directory for licenses. Is that by design? I have a Godep setup to handle deps and I would like to ensure (without legal status) that the included deps are of a certain license.

I am now using a series of bash commands to extract the deps as seen by `dep` but it is a bit shaky.

**Comment by vmarkovtsev** [comment]
Yes, this is by design. You need to compile a list of project directories and feed it to GLD. E.g. try this:

```
find vendor/ -mindepth 2 -maxdepth 2|xargs license-detector
```

**Comment by dahankzter** [comment]
Ok I will use some heuristics to find the project directories. Would it be possible to extend the tools scope to support vendor/modules etc recursively?

**Comment by vmarkovtsev** [comment]
I guess it will be possible by adding an extra command which runs those heuristics and outputs the list of probable project directories to stdout. GLD's algorithm is designed to be project-centric, that's the axis on which everything else is attached.
